api/rest/bundle_handler.go

```diff
@@ -56,14 +56,31 @@ type realClock struct{}
 
 func (realClock) Now() time.Time { return time.Now() }
 
-func NewBundleHandler(workDir string, collectors []collector.Collector, timeout time.Duration) BundleHandler {
-	return BundleHandler{
+func NewBundleHandler(workDir string, collectors []collector.Collector, timeout time.Duration) (*BundleHandler, error) {
+	f, err := os.Stat(workDir)
+	if err != nil {
```

Comment:
Not sure what you guys think, but in Go, I tend to think of `else` and `else if` statements as mild code smells, and usually can find a way to work around it. In this particular case, and the case below on line 72, the else statement doesn't look to be needed at all. I personally find it a bit more readable, since you don't need to hold quite as much branching logic in mind.

```suggestion
	if err != nil {
		if os.IsNotExist(err) {
			err = os.Mkdir(workDir, dirPerm)
			if err != nil {
				logrus.WithError(err).Errorf("workDir does not exist and could not be created %s", workDir)
				return nil, err
			}
		} 
		logrus.Errorf("Could not open workDir %s", workDir)
		return nil, err
	}
		
	if f.Mode().IsRegular() {
		logrus.Errorf("workDir exists but is not a directory: %s", workDir)
		return nil, err
	}
```

Reply:
+1 https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
I concur. I also like to avoid multiple indentation. Here if we invert `os.IsNotExist` we can have even simpler code
```go
	if err != nil {
		if !os.IsNotExist(err) {
			return nil, err
		}
		
		err = os.Mkdir(workDir, dirPerm)
		if err != nil {
			return nil, err
		}
	}

	if f.Mode().IsRegular() {
		return nil, err
	}
```